knowledge-base/prevent-text-cut-off-pdf-conversion-radwordsprocessing.md

```diff
@@ -0,0 +1,98 @@
+
+    ```csharp
+    internal class FontsProvider : Telerik.Windows.Documents.Extensibility.FontsProviderBase
+    {
+        private readonly string fontFolder = Environment.GetFolderPath(Environment.SpecialFolder.Fonts);
+
+        public override byte[] GetFontData(Telerik.Windows.Documents.Core.Fonts.FontProperties fontProperties)
+        {
+            string fontFamilyName = fontProperties.FontFamilyName;
+            bool isBold = fontProperties.FontWeight == Telerik.Documents.Core.Fonts.FontWeights.Bold;
+            if (fontFamilyName == "Tahoma" && isBold)
+            {
+                return GetFontDataFromFontFolder("tahomabd.ttf");
+            }
+            else if (fontFamilyName == "Tahoma")
+            {
+                return GetFontDataFromFontFolder("tahoma.ttf");
+            }
+            else if (fontFamilyName == "Arial" && isBold)
+            {
+                return GetFontDataFromFontFolder("arialbd.ttf");
+            }
+            else if (fontFamilyName == "Arial")
+            {
+                return GetFontDataFromFontFolder("arial.ttf");
+            }
+            else if (fontFamilyName == "Segoe UI")
+            {
+                return GetFontDataFromFontFolder("segoeui.ttf");
+            }
+
+            return null;
+        }
+
+        private byte[] GetFontDataFromFontFolder(string fontFileName)
+        {
+            using (FileStream fileStream = File.OpenRead(this.fontFolder + "\\" + fontFileName))
+            {
+                using (MemoryStream memoryStream = new MemoryStream())
+                {
+                    fileStream.CopyTo(memoryStream);
+                    return memoryStream.ToArray();
+                }
+            }
+        }
+    }
+    ```
+
+2. Before converting your HTML document to PDF, set the custom `FontsProvider` to the `FontsProvider` property of the `FixedExtensibilityManager`.
+
+    ```csharp
+    Telerik.Windows.Documents.Extensibility.FontsProviderBase fontsProvider = new FontsProvider();
+    Telerik.Windows.Documents.Extensibility.FixedExtensibilityManager.FontsProvider = fontsProvider;
+    ```
+
+Following these steps will ensure access to the necessary font data, preventing text from being cut off and ensuring all characters, including those with special characters, are properly rendered in the PDF document.
+
+## See Also
+- [PDF Format Provider]({%slug radwordsprocessing-formats-and-conversion-pdf-pdfformatprovider%})
+- [Fonts in RadPdfProcessing]({%slug radpdfprocessing-concepts-fonts%})
+- [How to Implement a FontsProvider]({%slug pdfprocessing-implement-fontsprovider%})
+- [RadWordsProcessing Documentation]({%slug radwordsprocessing-overview%})
+- [Preserving the Font in PDF Export from Excel]({%slug preserve-font-boldness-pdf-export-radspreadprocessing%})
```

Comment:
reference KB in another article (maybe "Fonts" or "FontsProvider implementation")